src/main/scala/firrtl/transforms/Flatten.scala

```diff
@@ -36,55 +43,47 @@ class Flatten extends InlineInstances {
        }
      }
 
-   // Replicate the circuit such that we inline only the modules that are below hierarchy given by the annotations
-   // Along with replication we rewrite the original circuit to refer to the new modules that will be inlined
+   /**
+    *  Modifies the circuit by replicating the hierarchy under the annotated objects (mods and insts) and
+    *  by rewriting the original circuit to refer to the new modules that will be inlined later
+    */
    def duplicateSubCircuitsFromAnno(c : Circuit, mods:  Set[ModuleName], insts : Set[ComponentName], modsToInline : mutable.Set[ModuleName]) : Circuit = {
```

Comment:
Instead of using a mutable.Set, you can return the tuple (Circuit, Set[ModuleName]), e.g.
```scala
def duplicateSubCircuitsFromAnno(c : Circuit, mods:  Set[ModuleName], insts : Set[ComponentName]) : (Circuit, Set[ModuleName]) = ...
...
val (newCircuit, modsToInline) = duplicateSubCircuitsFromAnno(c, mods, insts)
```